reason-parser/src/reason_parser.mly

```diff
@@ -1200,7 +1201,7 @@ _use_file:
 parse_core_type:
     core_type EOF
       {
-        let core_loc = mklocation $startpos($1) $endpos($1) in
+        let core_loc = mklocation $symbolstartpos $endpos($1) in
```

Comment:
Yes, there is a difference. From the Menhir manual:
```
The position $startpos is sometimes “further towards the left” than one would like. For example, in the
following production:
declaration: modifier? variable { $startpos }
the keyword $startpos represents the start position of the optional modifier modifier?. If this modifier
turns out to be absent, then its start position is (by definition) the end position of the most recently parsed
symbol. This may not be what is desired: perhaps the user would prefer in this case to use the start position
of the symbol variable. This is achieved by using $symbolstartpos instead of $startpos
```